File: packages/posthog/ios/Plugin/Classes/Options/SessionReplayOptions.swift
import Foundation

@objc public class SessionReplayOptions: NSObject {
    private var screenshotMode: Bool?
    private var maskAllTextInputs: Bool?
    private var maskAllImages: Bool?
    private var maskAllSandboxedViews: Bool?
    private var captureNetworkTelemetry: Bool?
    private var debouncerDelay: Double?

    init(screenshotMode: Bool? = false, maskAllTextInputs: Bool? = true, maskAllImages: Bool? = true, maskAllSandboxedViews: Bool? = true, captureNetworkTelemetry: Bool? = true, debouncerDelay: Double? = 1.0) {
        self.screenshotMode = screenshotMode
        self.maskAllTextInputs = maskAllTextInputs
        self.maskAllImages = maskAllImages
        self.maskAllSandboxedViews = maskAllSandboxedViews
        self.captureNetworkTelemetry = captureNetworkTelemetry
        self.debouncerDelay = debouncerDelay
    }

    func getScreenshotMode() -> Bool? {
        return screenshotMode
    }

    func getMaskAllTextInputs() -> Bool? {
        return maskAllTextInputs
    }

    func getMaskAllImages() -> Bool? {
        return maskAllImages
    }

    func getMaskAllSandboxedViews() -> Bool? {
        return maskAllSandboxedViews
    }

    func getCaptureNetworkTelemetry() -> Bool? {
        return captureNetworkTelemetry
    }

    func getDebouncerDelay() -> Double? {
        return debouncerDelay
    }
}

File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/SessionReplayOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.Nullable;

public class SessionReplayOptions {

    @Nullable
    private Boolean screenshotMode = false;

    @Nullable
    private Boolean maskAllTextInputs = true;

    @Nullable
    private Boolean maskAllImages = true;

    @Nullable
    private Boolean maskAllSandboxedViews = true;

    @Nullable
    private Boolean captureNetworkTelemetry = true;

    @Nullable
    private Double debouncerDelay = 1.0;

    public SessionReplayOptions() {}

    public SessionReplayOptions(
        @Nullable Boolean screenshotMode,
        @Nullable Boolean maskAllTextInputs,
        @Nullable Boolean maskAllImages,
        @Nullable Boolean maskAllSandboxedViews,
        @Nullable Boolean captureNetworkTelemetry,
        @Nullable Double debouncerDelay
    ) {
        this.screenshotMode = screenshotMode;
        this.maskAllTextInputs = maskAllTextInputs;
        this.maskAllImages = maskAllImages;
        this.maskAllSandboxedViews = maskAllSandboxedViews;
        this.captureNetworkTelemetry = captureNetworkTelemetry;
        this.debouncerDelay = debouncerDelay;
    }

    @Nullable
    public Boolean getScreenshotMode() {
        return screenshotMode;
    }

    @Nullable
    public Boolean getMaskAllTextInputs() {
        return maskAllTextInputs;
    }

    @Nullable
    public Boolean getMaskAllImages() {
        return maskAllImages;
    }

    @Nullable
    public Boolean getMaskAllSandboxedViews() {
        return maskAllSandboxedViews;
    }

    @Nullable
    public Boolean getCaptureNetworkTelemetry() {
        return captureNetworkTelemetry;
    }

    @Nullable
    public Double getDebouncerDelay() {
        return debouncerDelay;
    }
}
